src/contexts/login-context.js

```diff
@@ -0,0 +1,93 @@
+import axios from "axios";
+import { createContext, useContext, useReducer, useState } from "react";
+import { useNavigate } from "react-router-dom";
+
+const LoginContext = createContext("");
+const defaultCredentialTemplate = {
+    email: "",
+    password: ""
+}
+
+const userInfoTemplate = {
+    details: "",
+    encodedToken: ""
+};
+
+const loginReducer = (userCredentials, action) => {
+    switch(action.type) {
+        case "EMAIL" :
+            return {...userCredentials, email : action.payload}
+        case "PASSWORD" : 
+            return {...userCredentials, password : action.payload}
+        default: return ({...defaultCredentialTemplate})
+    }
+}
+
+const userInfoReducer = (userInfo, action) => {
+    switch(action.type) {
+        case "DETAILS" : 
+            return {...userInfo, details: action.payload}
+        case "ENCODED_TOKEN" :
+            return {...userInfo, encodedToken: action.payload}
+        default: return ({...userInfoTemplate})
+    }
+}
+
+const LoginContextProvider = ({children}) => {
+    const [userCredentials, loginDispatch] = useReducer(loginReducer, defaultCredentialTemplate);
+    const [isUserLoggedIn , setUserLoggedIn] = useState(false);
+    const [userInfo, userInfoDispatch] = useReducer(userInfoReducer, userInfoTemplate);
+    const [validationMessage, setValidationMessage] = useState("");
+    const navigate = useNavigate();
+
+    const loginHandler = async (e, creds) => {
+        e.preventDefault();
+        try {
+            const emptyFields = Object.keys(creds).filter(attribute => creds[attribute] === "");
```

Comment:
This seems a new approach along with the reducers you have developed for login and signup actions. Thanks @anushasunil diving deep from your actions